python/cudf/cudf/core/dataframe.py

```diff
@@ -6533,12 +6533,14 @@ def to_struct(self, name=None):
         field_names = [str(name) for name in self._data.names]
 
         col = cudf.core.column.build_struct_column(
-            names=field_names, children=self._data.columns, size=len(self)
+            names=field_names,
+            children=tuple(
+                [col.copy(deep=True) for col in self._data.columns]
```

Comment:
Seems like the docstring claiming a copy of the columns was made was previously a lie?

Reply:
See below.